### PR TITLE
*: add GetArgs interface

### PR DESCRIPTION
**Pull request by hanfei1991**
prepare to refactor function model.
@shenli @coocood @zimulala @tiancaiamao PTAL

### CODE REVIEW

File: plan/expression_rewriter.go [hunk 1]
```diff
@@ -452,7 +451,15 @@ func (er *expressionRewriter) Leave(inNode ast.Node) (retNode ast.Node, ok bool)
 	case *ast.CaseExpr:
 		er.caseToExpression(v)
 	case *ast.FuncCastExpr:
-		er.castToScalarFunc(v)
+		arg := er.ctxStack[len(er.ctxStack)-1]
+		er.checkArgsOneColumn(arg)
```
Comment (coocood) [inline plan/expression_rewriter.go]: Why not put this check into `NewCastFunc`?

Comment (hanfei1991) [inline plan/expression_rewriter.go]: NewCastFunction can't get the er ref.

**Comment by shenli** [comment]
LGTM

File: plan/expression_rewriter.go (hunk 1 above)
Comment (coocood) [inline plan/expression_rewriter.go]: `checkArgsOneColumn` doesn't need `er`.

Comment (hanfei1991) [inline plan/expression_rewriter.go]: checkArgsOneColumn only used in this file.

**Pull request merged**